server/jetstream_cluster_4_test.go

```diff
@@ -2549,3 +2549,294 @@ func TestJetStreamClusterMetaSyncOrphanCleanup(t *testing.T) {
+			func(t *testing.T) {
+				// This is the success condition for all sub-tests below
+				var ExpectedMsgId = ""
+				checkMessage := func(t *testing.T, msg *nats.Msg) {
+					msgMeta, err := msg.Metadata()
+					require_NoError(t, err)
+
+					// Check sequence number
+					require_Equal(t, msgMeta.Sequence.Stream, ChosenSeq)
+
+					// Check message id
+					require_NotEqual(t, ExpectedMsgId, "")
+					require_Equal(t, msg.Header.Get(nats.MsgIdHdr), ExpectedMsgId)
+				}
+
+				checkRawMessage := func(t *testing.T, msg *nats.RawStreamMsg) {
+					// Check sequence number
+					require_Equal(t, msg.Sequence, ChosenSeq)
+
+					// Check message id
+					require_NotEqual(t, ExpectedMsgId, "")
+					require_Equal(t, msg.Header.Get(nats.MsgIdHdr), ExpectedMsgId)
+				}
+
+				// Setup: start server or cluster, connect client
+				var server *Server
+				if ClusterSize == 1 {
+					server = RunBasicJetStreamServer(t)
+					defer server.Shutdown()
+				} else {
+					c := createJetStreamCluster(t, jsClusterTempl, "HUB", _EMPTY_, ClusterSize, 22020, true)
+					defer c.shutdown()
+					server = c.randomServer()
+				}
+
+				// Setup: connect
+				var nc *nats.Conn
+				var js nats.JetStreamContext
+				nc, js = jsClientConnect(t, server)
+				defer nc.Close()
+
+				// Setup: create stream
+				_, err := js.AddStream(&nats.StreamConfig{
+					Replicas: R,
+					Name:     StreamName,
+					Subjects: []string{StreamSubject},
+				})
+				require_NoError(t, err)
+
+				// Setup: populate stream
+				buf := make([]byte, 100)
+				for i := uint64(1); i <= NumMessages; i++ {
+					msgId := nuid.Next()
+					pubAck, err := js.Publish(StreamSubjectPrefix+strconv.Itoa(int(i)), buf, nats.MsgId(msgId))
+					require_NoError(t, err)
+
+					// Verify assumption made in tests below
+					require_Equal(t, pubAck.Sequence, i)
+
+					if i == ChosenSeq {
+						// Save the expected message id for the chosen message
+						ExpectedMsgId = msgId
+					}
+				}
+
+				t.Run(
+					"Durable consumer bind",
+					func(t *testing.T) {
+
+						const Durable = "dlc"
+						c, err := js.AddConsumer(StreamName, &nats.ConsumerConfig{
+							Durable:       Durable,
+							DeliverPolicy: nats.DeliverByStartSequencePolicy,
+							OptStartSeq:   ChosenSeq,
+							Replicas:      R,
+						})
+						require_NoError(t, err)
+						defer func() {
+							err := js.DeleteConsumer(c.Stream, c.Name)
+							if err != nil {
+								t.Errorf("Error deleting consumer: %v", err)
+							}
+						}()
+
+						sub, err := js.PullSubscribe(
+							"",
+							"",
+							nats.Bind(StreamName, Durable),
+						)
+						require_NoError(t, err)
+						defer func() {
+							err := sub.Unsubscribe()
+							if err != nil {
+								t.Error("Error unsubscribing", err)
+							}
+						}()
+
+						msgs, err := sub.Fetch(1)
+						require_NoError(t, err)
+						require_Equal(t, len(msgs), 1)
+
+						checkMessage(t, msgs[0])
+					},
+				)
+
+				t.Run(
+					"Durable consumer",
```

Comment:
Minor nit I know but please make these subtest names CamelCase instead of with spaces.

Reply:
I will update this.

But out of curiosity, what's the reason? I see the codebase uses all kinds, spaces, dashes, camel.
Just trying to be consistent going forward?

Go itself has no issue with spaces:
```
=== RUN   TestJetStreamConsumeWithStartSequence
=== RUN   TestJetStreamConsumeWithStartSequence/servers=1,R=1
=== RUN   TestJetStreamConsumeWithStartSequence/servers=1,R=1/Durable_consumer_bind
=== RUN   TestJetStreamConsumeWithStartSequence/servers=1,R=1/Durable_consumer
=== RUN   TestJetStreamConsumeWithStartSequence/servers=1,R=1/Pull_consumer
=== RUN   TestJetStreamConsumeWithStartSequence/servers=1,R=1/Synchronous_consumer
=== RUN   TestJetStreamConsumeWithStartSequence/servers=1,R=1/Callback_subscribe
=== RUN   TestJetStreamConsumeWithStartSequence/servers=1,R=1/Channel_subscribe
=== RUN   TestJetStreamConsumeWithStartSequence/servers=1,R=1/Get_raw_stream_message
=== RUN   TestJetStreamConsumeWithStartSequence/servers=1,R=1/Get_last_message_by_subject
=== RUN   TestJetStreamConsumeWithStartSequence/servers=3,R=3
=== RUN   TestJetStreamConsumeWithStartSequence/servers=3,R=3/Durable_consumer_bind
=== RUN   TestJetStreamConsumeWithStartSequence/servers=3,R=3/Durable_consumer
=== RUN   TestJetStreamConsumeWithStartSequence/servers=3,R=3/Pull_consumer
=== RUN   TestJetStreamConsumeWithStartSequence/servers=3,R=3/Synchronous_consumer
=== RUN   TestJetStreamConsumeWithStartSequence/servers=3,R=3/Callback_subscribe
=== RUN   TestJetStreamConsumeWithStartSequence/servers=3,R=3/Channel_subscribe
=== RUN   TestJetStreamConsumeWithStartSequence/servers=3,R=3/Get_raw_stream_message
=== RUN   TestJetStreamConsumeWithStartSequence/servers=3,R=3/Get_last_message_by_subject
```